packages/marshal/src/marshal.js

```diff
@@ -216,28 +287,84 @@ export const makeMarshal = (
       return val;
     };
 
-    const fullRevive = makeDecodeFromCapData({
+    const decodeRemotableFromSmallcaps = stringEncoding => {
+      assert(stringEncoding.startsWith('$'));
+      // slots: $slotIndex.iface or $slotIndex
+      const i = stringEncoding.indexOf('.');
+      const index = Number(stringEncoding.slice(1, i));
+      // assert(i > 0, X`${rawTree} is not a valid slot encoding`);
+      // i < 0 means there was no iface included
+      const iface = i < 0 ? undefined : stringEncoding.slice(i + 1);
+      return unserializeSlot(index, iface);
```

Comment:
Won't `stringEncoding.slice(1, i)` chop off the last digit when `i` is -1 because `stringEncoding` doesn't contain a "."?
```suggestion
      const i = stringEncoding.indexOf('.');
      const slotIndex = Number(stringEncoding.slice(1, i < 0 ? Infinity : i));
      // i < 0 means there was no iface included.
      const iface = i < 0 ? undefined : stringEncoding.slice(i + 1);
      return unserializeSlot(slotIndex, iface);
```
Are we missing a slot deserialization test, or am I just wrong about the behavior of this code?

Reply:
You are correct that this is wrong. I was originally always including the "." I think.  I woudl use `undefined` rather than `Infinity`, but that's minor

Reply:
Took your next suggestion.